### PR TITLE
Feature/query

### LINKED ISSUE #8: Querying API
As a frontend engineer, I would like to be able to query the API using query params.

- [x] Create a way to query the API via any attribute
- [x] Some attributes may need to be prevented from querying, depending on authorisation/permissions
- [x] Test the querying

- [x] Eager/related queries should return only id and type

### PR DESCRIPTION
**Pull request by mosesokemwa**
Query GET route API via any attribute fro the front-end
Added modifiers in the models to reduce the load on returned data
Updated tests to match the included query attributes
Manual restricted query attributes(this will eventually be updated to match user role)

closes #8

### CODE REVIEW

**Review by Jakeii — COMMENTED** [review]
Does this handle multiple query params?

**Comment by Jakeii** [comment]
Could you test passing invalid query params too?

**Comment by mosesokemwa** [comment]
> Does this handle multiple query params?

According to the issue created, one should query the API via any attribute and not multiple params.
``learning?key1=val1``
``learning?key2=val2``

I'll update it to support
``learning?key1=val1&key2=val2``

And add a default sort query at the end
``learning?key1=val1&key2=val2&sort=xxxxx``

File: server/routes/activity.js [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 const router = new Router({
   prefix: '/activity'
 });
 
 
 
-router.get('/:id', async ctx => {
-  const activity = await Activity.query().where('id', ctx.params.id);
+router.get('/:id', queryStringSearch, async ctx => {
+  const activity = await Activity.query().where(ctx.query);
```
Comment (Jakeii) [inline server/routes/activity.js]: The queries need to be on the naked route, don't they? `/:id` will only return a single result, can't narrow that down with a query...

Comment (mosesokemwa) [reply server/routes/activity.js]: I realised that and I'm currently working on putting them on the naked route.

**Review by Jakeii — CHANGES_REQUESTED** [review]
if I pass a query parameter that doesn't exist I get a 500 error, should be a 400 with an "Invalid query parameter" message I think?

**Comment by Jakeii** [comment]
> if I pass a query parameter that doesn't exist I get a 500 error, should be a 400 with an "Invalid query parameter" message I think?

Still happening?

**Comment by mosesokemwa** [comment]
Could you share the steps to reproduce the error you are getting

**Pull request merged**